rudder-web/src/main/scala/com/normation/rudder/web/services/ComplianceData.scala

```diff
@@ -95,6 +95,163 @@ object RulePolicyMode {
+      case PolicyModeOverrides.Always =>
+        nodeMode match {
+          case Some(Audit) =>
+            (Audit.name,"<b>Audit</b> mode is overriden by this <i><b>Node</b></i>")
+          case nodeMode    =>
+
+            val (defaultMode,expl) = nodeMode match {
+              case Some(mode) =>
+                ( mode, s"<b>${mode.name}</b> mode is overriden by this <i><b>Node</b></i>")
+              case None =>
+                ( globalMode.mode, "This mode is the globally defined default. You can change it in the global <i><b>settings</b></i>.")
+            }
+            val default = (defaultMode.name,expl)
+
+            // We have a Set here so we only have 3 elem max (None, Some(audit), some(enforce))
+            directivesMode.toList match {
+              // No directives linked to the Rule, fallback to globalMode
+              case Nil => default
+              // We only have one element!!
+              case mode :: Nil => mode match {
+                case None => default
+                case Some(Audit) =>
+                  nodeMode match {
+                    case Some(Enforce) =>
+                     (Audit.name, "The <i><b>Node</b></i> is configured to <b class='text-Enforce'>enforce</b> but is overriden to <b>audit</b> by all <i><b>Directives</b></i>. ")
+                    case Some(Audit) =>
+                     (Audit.name, "<b>Audit</b> mode is overriden by both this <i><b>Node</b></i> and all <i><b>Directives</b></i>")
+                    case None =>
+                     (Audit.name,"<b>Audit</b> mode is overriden by all <i><b>Directives</b></i>")
+                  }
+                case Some(Enforce) =>
+                  nodeMode match {
+                    case Some(Audit) =>
+                     (Audit.name, "All <i><b>directives</b></i> are configured to <b class='text-Enforce'>enforce</b> but is overriden to <b>audit</b> by the <i><b>Node</b></i>. ")
+                    case Some(Enforce) =>
+                     (Enforce.name, "<b>Enforce</b> mode is overriden by both this <i><b>Node</b></i> and all <i><b>Directives</b></i>")
+                    case None =>
+                     (Enforce.name,"<b>Enforce</b> mode is overriden by all <i><b>Directives</b></i>")
+                  }
+
+              }
+
+              case modes  =>
+                // Here we now need to check if global mode is the same that all selected mode for directives
+                // So we replace any "None" (use global mode) by global mode and we will decide after what happens
+
+                directivesMode.map(_.getOrElse(defaultMode)).toList match {
+                  case Nil => default
+                  case mode :: Nil => (mode.name, s"<b>${mode.name}</b> mode is overriden by this <i><b>Node</b></i> and all <i><b>Directives</b></i>")
```

Comment:
why the logic is not the same as in previous section? 
And if node is enforce and you have defaultMode audit (overridable), and the directives are Audit + Default audit (overridable), then the text will be incorrect, as the Audit (overridable) will not by overriden by the node (because it is enforce).